### PR TITLE
feat(guards): ensure session data is real

### PR DESCRIPTION
**Pull request by jayceslesar**
Validate that session data (fake IP ect) can be cross referenced against the steam API to ensure this connection request is valid.

### CODE REVIEW

File: masterbase/guards.py [hunk 1]
```diff
@@ -68,3 +69,11 @@ async def session_closed_guard(connection: ASGIConnection, _: BaseRouteHandler)
 
 async def valid_session_guard(connection: ASGIConnection, _: BaseRouteHandler) -> None:
     """Validate session data is from a server we can check exists."""
+    fake_ip = connection.query_params["fake_ip"]
+    ip, fake_port = fake_ip.split(":")
+    converted_fake_ip = get_fake_ip(ip)
```
Comment (lili7h) [inline masterbase/guards.py:74]: Not knowing how the fake IP conversion works (i read the function, looks wack), these variable names are confusing. We have a fake_ip already from the connection, apparently. Then we split out a real ip, then get a fake ip from it again? Whats happening here

**Review by lili7h — COMMENTED** [review]
looks good to me, useful stuff

File: masterbase/guards.py (hunk 1 above)
Comment (jayceslesar) [reply masterbase/guards.py:74]: yeah the fake_ip in the DB is what is returned from the client, and then some sketchy math gets done to convert it into the IP/format (IDK really) that the steam API expects

Comment (jayceslesar) [reply masterbase/guards.py:74]: I think its actually a universe ID or something I dont really remember

**Comment by lili7h** [comment]
I like the new names better

**Pull request merged**